src/sailfishui/plugin/PrivacyNotice.qml

```diff
@@ -28,13 +28,17 @@ Item {
 
     property Page page
     readonly property bool pageActive: page.status === PageStatus.Active
+    property bool noticeGiven: false
 
     onPageActiveChanged: {
         if (pageActive) {
             if (PrivacySettings.privacyNoticeAccepted) {
                 pageStack.pop()
+            } else if(! noticeGiven) {
+                noticeGiven = true
+                pageStack.push(privacyNoticeDialog, null, PageStackAction.Immediate)
             } else {
-                pageStack.push(privacyNoticeDialog)
+                pageStack.pop()
```

Comment:
This is more like a safety guard, right?

Reply:
This is what pops back to the settings menu if the PrivacyNotice isn't accepted.

Reply:
Yeah, checked that as well. I was surprised that this PrivacyNotice was an item not Dialog. Well... works like this as well. This crash reporter has not seen that much love lately.